### ISSUE TITLE
Type context not respected when passing list literal to `Sequence` parameter

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#1699]
> @sharkdp I have a similar problem trying to use ty with DRF typings that started from 27 alpha, should I create a separate issue for that? 
> 
> ```py
> from __future__ import annotations
> 
> from typing import TYPE_CHECKING, Literal
> 
> if TYPE_CHECKING:
>     from collections.abc import Sequence
> 
> 
> type _Method = Literal["GET", "POST", "PUT", "DELETE"]
> 
> 
> def test(method: Sequence[_Method]) -> None:
>     return None
> 
> 
> test(["POST"])
> ```
> 
> 
> ```bash
> uvx ty check  a.py
> error[invalid-argument-type]: Argument to function `test` is incorrect
>   --> a.py:16:6
>    |
> 16 | test(["POST"])
>    |      ^^^^^^^^ Expected `Sequence[_Method]`, found `list[Unknown | str]`
>    |
> info: Function defined here
>   --> a.py:12:5
>    |
> 12 | def test(method: Sequence[_Method]) -> None:
>    |     ^^^^ ------------------------- Parameter declared here
> 13 |     return None
>    |
> info: rule `invalid-argument-type` is enabled by default
> 
> Found 1 diagnostic
> ``` 

 _Originally posted by @kalekseev in [#1689](https://github.com/astral-sh/ty/issues/1689#issuecomment-3595518452)_

**Comment by AlexWaygood** [comment]
This looks like it's the same as https://github.com/astral-sh/ty/issues/1576

**Comment by sharkdp** [comment]
Looks like this has the same root cause as https://github.com/astral-sh/ty/issues/1576?
